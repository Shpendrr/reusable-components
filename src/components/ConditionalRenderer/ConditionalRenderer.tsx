import { FC, ReactNode } from "react";

interface IConditionalRendererProps {
  condition: boolean;
  render: () => ReactNode;
}

const ConditionalRenderer: FC<IConditionalRendererProps> = ({
  condition,
  render,
}) => {
  return condition ? render() : null;
};

export default ConditionalRenderer;
