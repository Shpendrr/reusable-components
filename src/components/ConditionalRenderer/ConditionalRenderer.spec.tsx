import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import ConditionalRenderer from "@/components/ConditionalRenderer/ConditionalRenderer";

describe("ConditionalRenderer", () => {
  it("renders the content when the condition is true", () => {
    const { getByText } = render(
      <ConditionalRenderer
        condition={true}
        render={() => <div>Rendered content</div>}
      />
    );
    expect(getByText("Rendered content")).toBeInTheDocument();
  });

  it("does not render the content when the condition is false", () => {
    const { queryByText } = render(
      <ConditionalRenderer
        condition={false}
        render={() => <div>Rendered content</div>}
      />
    );
    expect(queryByText("Rendered content")).not.toBeInTheDocument();
  });

  it("matches the snapshot", () => {
    const { container } = render(
      <ConditionalRenderer
        condition={true}
        render={() => <div>Rendered content</div>}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
