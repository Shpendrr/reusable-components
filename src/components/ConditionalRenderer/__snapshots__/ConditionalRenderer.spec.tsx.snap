// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConditionalRenderer matches the snapshot 1`] = `
<div>
  Rendered content
</div>
`;
